datasetIngestor/scp.go

```diff
@@ -32,7 +32,17 @@ func keyString(k ssh.PublicKey) string {
 	return k.Type() + " " + base64.StdEncoding.EncodeToString(k.Marshal()) // e.g. "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTY...."
 }
 
+/* trustedHostKeyCallback returns a function that serves as a callback for SSH host key verification.
+If a trustedKey is provided, the callback will verify if the key from the server matches the trustedKey.
+If they don't match, it returns an error.
+If no trustedKey is provided, the callback will log a warning that SSH-key verification is not in effect,
+but it will not stop the connection.
+Parameters:
+trustedKey: A string representation of the trusted SSH public key.
+Returns:
+An ssh.HostKeyCallback function for SSH host key verification. */
 func trustedHostKeyCallback(trustedKey string) ssh.HostKeyCallback {
```

Comment:
is this function ever used? I am struggling to find where...

Reply:
Not really

Reply:
ok, I struggled to understand how it could be used as the ssh private key should not be in the cli. Happy to merge!